src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -308,5 +308,135 @@ void IslandWindow::UpdateTheme(const winrt::Windows::UI::Xaml::ElementTheme& req
     ::InvalidateRect(_window.get(), nullptr, false);
 }
 
+// Method Description:
+// - Toggles our fullscreen state. See _SetIsFullscreen for more details.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void IslandWindow::ToggleFullscreen()
+{
+    _SetIsFullscreen(!_fullscreen);

```

Comment:
It mildly bothers me that there's theoretically a possibility for `IslandWindow`'s `_fullscreen` variable to be out of sync with `TerminalApp`'s `_fullscreen` variable because it's stored in two places. Any way to avoid it?

Reply:
Probably not as cleanly as I'd like. If we exposed it simply through a winrt method on `TerminalPage`, then we'd have to make a binary boundary hop every time we wanted to query it.
`NonClientIsland` uses that state during `_UpdateDragRegion` and `WM_PAINT`, and `_UpdateDragRegion` is called _every time the window moves_. So I don't really want to have to dll hop there.

Reply:
Makes sense. OK, leave it.